rsocket-core/src/main/java/io/rsocket/LeaseSocketAcceptor.java

```diff
@@ -0,0 +1,8 @@
+package io.rsocket;
+
+import reactor.core.publisher.Mono;
+
+public interface LeaseSocketAcceptor {
```

Comment:
Last time leasing was implemented it was leaky, and there was a bunch of different "leasing" acceptors and sockets. I don't want that to happen again. I don't want special sockets for leasing. The person using it shoudn't know that it's happening other than a flag in setup.

Reply:
Users of requester RSocket would need to know when lease is granted/withdrawn, and how many credits are available. Otherwise users would blindly retry requests failed due to missing lease, which is effectively polling. Also, It can be holder for "simple" RSocket, and whatever factory wants to expose to user (Context). Or raw socket is enough?

Reply:
There is availability in the RSocket - you need to update that to 0 when there are no leases, and then it's up to the client to decided with to send the request or not. For instance the load balancer uses availability wither or not to send the request to that RSocket.

Reply:
Actually what we ended up doing was time as make availability a ratio of available_leases/total_leases  - servers with more leases should be more available

Reply:
With this change, would we also collapse RSocketFactory.leaseAcceptor and acceptor?  This bit confused me when trying to use the API.

Reply:
@yschimke I am reworking this atm based on Robert comments -  there will be no leaseAcceptor, and lease changes will be reflected in RSocket availability. From user point of view  `Client/ServerRSocketFactory` will be augmented with `enableLease(something)`. Hopefully will come up with runnable example by the end of weekend